custom_components/isy994/helpers.py

```diff
@@ -423,3 +424,17 @@ def convert_isy_value_to_hass(
     if fallback_precision:
         return round(float(value), fallback_precision)
     return value
+
+
+def _map_folders_for_nodes(
```

Comment:
Don't know if it makes it easier, but you can grab the folder from the path using the nodes iterator.

```python
for (path, node) in isy.nodes:
  top_folder = path.split('/')[1]

  # If device is not in a folder, this will be the node name
  if top_folder == node.name:
    top_folder = ""

```

Reply:
That's a bit easier :)

Will try to refactor before the flight takes off